superset-frontend/src/explore/components/DataTableControl/useTableColumns.test.ts

```diff
@@ -113,87 +154,45 @@ test('use only the first record columns', () => {
     },
     {
       Cell: expect.any(Function),
-      Header: 'col03',
-      accessor: expect.any(Function),
-      id: 'col03',
-    },
-    {
-      Cell: expect.any(Function),
-      Header: asciiKey,
-      accessor: expect.any(Function),
-      id: asciiKey,
-    },
-    {
-      Cell: expect.any(Function),
-      Header: unicodeKey,
-      accessor: expect.any(Function),
-      id: unicodeKey,
-    },
-  ]);
-
-  hook.result.current.forEach((col: JsonObject) => {
-    expect(col.accessor(newData[0])).toBe(newData[0][col.Header]);
-  });
-
-  hook.result.current.forEach((col: JsonObject) => {
-    expect(col.Cell({ value: newData[0].col01 })).toBe(BOOL_TRUE_DISPLAY);
-    expect(col.Cell({ value: newData[0].col02 })).toBe(BOOL_FALSE_DISPLAY);
-    expect(col.Cell({ value: newData[0].col03 })).toBe('secret');
-    expect(col.Cell({ value: newData[0][asciiKey] })).toBe(asciiKey);
-    expect(col.Cell({ value: newData[0][unicodeKey] })).toBe(unicodeKey);
-  });
```

Comment:
This logic was incorrect, and caused each cell formatter to process each value on the row